swarm_copy/app/routers/threads.py

```diff
@@ -0,0 +1,108 @@
+"""Threads CRUDs."""
+
+import json
+import logging
+from typing import Annotated
+
+from fastapi import APIRouter, Depends
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from swarm_copy.app.database.db_utils import get_thread
+from swarm_copy.app.database.schemas import MessagesRead, ThreadsRead
+from swarm_copy.app.database.sql_schemas import Messages, Threads
+from swarm_copy.app.dependencies import get_session, get_user_id
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter(prefix="/threads", tags=["Threads' CRUD"])
+
+
+@router.post("/")
+async def create_thread(
+    session: Annotated[AsyncSession, Depends(get_session)],
+    user_id: Annotated[str, Depends(get_user_id)],
+    title: str = "title",
+) -> ThreadsRead:
+    """Create thread."""
+    new_thread = Threads(user_id=user_id, title=title)
+    session.add(new_thread)
+    await session.commit()
+    await session.refresh(new_thread)
+
+    return ThreadsRead(**new_thread.__dict__)
+
+
+@router.get("/")
+async def get_threads(
+    session: Annotated[AsyncSession, Depends(get_session)],
+    user_id: Annotated[str, Depends(get_user_id)],
+) -> list[ThreadsRead]:
+    """Get threads for a user."""
+    thread_result = await session.execute(
+        select(Threads).where(Threads.user_id == user_id)
+    )
+    threads = thread_result.scalars().all()
+    return [ThreadsRead(**thread.__dict__) for thread in threads]
+
+
+@router.get("/{thread_id}")
+async def get_messages(
```

Comment:
Same, no strong opinions, but for the moment we don't want to do any operations on single messages except for getting them. If we add more it would make more sense